bpf/bpf_sock.c

```diff
@@ -154,12 +154,13 @@ static __always_inline int sock4_update_revnat(struct bpf_sock_addr *ctx,
 	return ret;
 }
 
-static __always_inline int sock4_delete_revnat(struct bpf_sock *ctx)
+static __always_inline int sock4_delete_revnat(struct bpf_sock *ctx,
+					       struct bpf_sock *ctx_full)
```

Comment:
Nit: this is confusing. When I see `ctx` I expect its a usable context. I would make the first argument `ctx_fake` and the second `ctx`. Hopefully that hints that the first isn't usable.

Better yet, change to `int sock4_delete_revnat(struct bpf_sock *ctx, __u32 address, __u16 port);` since most of the fake ctx is uninitialized anyway.